src/backend/distributed/commands/table.c

```diff
@@ -939,6 +939,32 @@ PreprocessAlterTableStmt(Node *node, const char *alterTableCommand,
 				 */
 				constraint->skip_validation = true;
 			}
+			else if (constraint->contype == CONSTR_PRIMARY)
+			{
+				if (constraint->conname == NULL)
+				{
+					Relation rel = RelationIdGetRelation(leftRelationId);
+					constraint->conname = ChooseIndexName(RelationGetRelationName(rel),
```

Comment:
Note that here you are actually changing the original command because you are directly modifying the original object. Then you are also modifying the `newcmd` which you pass to the deparser. I think you can should directly modify the conname in `newcmd` and not touch the original command.

Reply:
I change the original command intentionally. The idea is that we should run the same command,  with the name we explicitly created, across all the tables on the coordinator as well as the workers.  Hence I change the original command so that the standard utility function uses the new name. Is there any downside to change the original command?

Reply:
We generally don't want to change the original command that is run on the coordinator, we want to let Postgres do its thing on the table that is on the coordinator. The behavior of this command may change and it's more difficult to maintain it. This is the reason why we create an alter table command deep copy at the beginning.

Reply:
> We generally don't want to change the original command that is run on the coordinator, we want to let Postgres do its thing on the table that is on the coordinator. 

In this case, wouldn't we want to make sure that the internal utility hook uses the index name we choose here? We do something similar for CREATE INDEX without name.

Reply:
> We do something similar for CREATE INDEX without name.

Hmm, we should also keep that in mind in case the behavior changes in the future.

Reply:
I am not sure about this.

However if we decide to directly modify the original command I guess there is no need to modify newCmd as well.

Reply:
Refactoring the code, we are not chaning newCmd anymore.